larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h

```diff
@@ -0,0 +1,393 @@
+     *
+     *  @param  discreteProbabilityVector a discrete probability vector to resample from
+     *  @param  resamplingPoints the points to resample the discrete probability vector with 
+     */
+    DiscreteProbabilityVector(const DiscreteProbabilityVector &discreteProbabilityVector, const ResamplingPoints &resamplingPoints);
+
+    /**
+     *  @brief  Constructor
+     *
+     *  @param  discreteProbabilityVector a discrete probability vector to randomly rearrange
+     *  @param  randomNumberGenerator the random number generator for the random reshuffling
+     */
+    DiscreteProbabilityVector(const DiscreteProbabilityVector &discreteProbabilityVector, std::mt19937 &randomNumberGenerator);
+
+    /**
+     *  @brief  Evaluate the cumulative probability at arbitrary x
+     *
+     *  @param  x the x value
+     *
+     *  @return the cumulative probability
+     */
+    float EvaluateCumulativeProbability(const float x) const;
+
+    /**
+     *  @brief  Get the size of the probability vector
+     *
+     *  @return the probability vector size
+     */
+    size_t GetSize() const;
+
+    /**
+     *  @brief  Get the x value of the element in the vector
+     *
+     *  @param  index the index in the vector
+     *
+     *  @return the x value
+     */
+    float GetX(const size_t index) const;
+
+    /**
+     *  @brief  Get the probability value of the element in the vector
+     *
+     *  @param  index the index in the vector
+     *
+     *  @return the probablity
+     */
+    float GetProbability(const size_t index) const;
+
+    /**
+     *  @brief  Get the probability density value of the element in the vector
+     *
+     *  @param  index the index in the vector
+     *
+     *  @return the probablity density
+     */
+    float GetProbabilityDensity(const size_t index) const;
+
+    /**
+     *  @brief  Get the cumulative probability value of the element in the vector
+     *
+     *  @param  index the index in the vector
+     *
+     *  @return the cumulative probability
+     */
+    float GetCumulativeProbability(const size_t index) const;
+
+    /**
+     *  @brief  Get the width of the element in the vectorr
+     *
+     *  @param  index the index in the vector
+     *
+     *  @return the width of the probability bin
+     */
+    float GetWidth(const size_t index) const;
+
+    /**
+     *  @brief  Get all information stored at a particular index
+     *
+     *  @param  index the index in the vector
+     *  @param  x the x value
+     *  @param  probabilityDensity the probability density value
+     *  @param  cumulativeProbability the cumulative probability value
+     *  @param width the width of the probability bin
+     */
+    void GetAllAtIndex(const size_t index, float &x, float &probabilityDensity, float &cumulativeProbability, float &width) const;
+
+private:
+
+    /**
+     *  @brief  DiscreteProbabilityData class
+     */
+    class DiscreteProbabilityDatum
+    {
+        public:
```

Comment:
Check indentation of public/private is consistent with rest of codebase.

Reply:
It is consistent now